x-pack/plugins/security_solution/server/usage/detections/detections_helpers.ts

```diff
@@ -301,3 +483,116 @@ export const getMlJobMetrics = async (
 
   return [];
 };
+
+export const getDetectionRuleMetrics = async (
+  kibanaIndex: string,
+  signalsIndex: string,
+  esClient: ElasticsearchClient,
+  savedObjectClient: SavedObjectsClientContract
+): Promise<DetectionRuleAdoption> => {
+  let rulesUsage: DetectionRulesTypeUsage = initialDetectionRulesUsage;
+  const ruleSearchOptions: RuleSearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: [],
+    ignoreUnavailable: true,
+    index: kibanaIndex,
+    size: 10_000,
+  };
+
+  let detectionAlertsResp: AlertsAggregationResponse | undefined;
+
+  try {
+    const { body: ruleResults } = await esClient.search<RuleSearchResult>(ruleSearchOptions);
+
+    // @ts-expect-error `SearchResponse['hits']['total']` incorrectly expects `number` type instead of `{ value: number }`.
```

Comment:
Since `ts-expect-error` is dangerous in that anyone can break this code with any invalid typescript once it's been added, shouldn't we use just an as cast like this?

```ts
    ({ body: detectionAlertsResp } = (await esClient.search({
      index: signalsIndex,
      size: 0,
      body: {
        aggs: {
          detectionAlerts: {
            terms: { field: 'signal.rule.id.keyword' },
          },
        },
        query: {
          bool: {
            filter: [
              {
                range: {
                  '@timestamp': {
                    gte: 'now-24h',
                    lte: 'now',
                  },
                },
              },
            ],
          },
        },
      },
    })) as { body: AlertsAggregationResponse });
```

I don't like `as` casts either, but with the ts-expect-error I could change that code to anything compared to an as cast I at least will get some type checking happening with syntax and other things.

Reply:
Thanks! Updated in 0c91a78